Example/Messaging/Sample/Sample/SceneDelegate.swift

```diff
@@ -22,28 +22,37 @@ import FirebaseInstallations
 class SceneDelegate: UIResponder, UIWindowSceneDelegate, MessagingDelegate {
   var window: UIWindow?
   let identity = Identity()
+  let settings = UserSettings()
   var cancellables = Set<AnyCancellable>()
 
   func scene(_ scene: UIScene, willConnectTo session: UISceneSession,
              options connectionOptions: UIScene.ConnectionOptions) {
     let contentView = ContentView()
+    if settings.shouldUseDelegateThanNotification {
+      Messaging.messaging().delegate = self
+    }
     // Use a UIHostingController as window root view controller.
     if let windowScene = scene as? UIWindowScene {
       let window = UIWindow(windowScene: windowScene)
       window
-        .rootViewController = UIHostingController(rootView: contentView.environmentObject(identity))
+        .rootViewController = UIHostingController(rootView: contentView
+          .environmentObject(identity)
+          .environmentObject(settings)
+        )
 
       self.window = window
       window.makeKeyAndVisible()
     }
 
-    // Subscribe to token refresh
-    _ = NotificationCenter.default
-      .publisher(for: Notification.Name.MessagingRegistrationTokenRefreshed)
-      .map { $0.object as? String }
-      .receive(on: RunLoop.main)
-      .assign(to: \Identity.token, on: identity)
-      .store(in: &cancellables)
+    if !settings.shouldUseDelegateThanNotification {
+      // Subscribe to token refresh
+      _ = NotificationCenter.default
```

Comment:
I think since it's stored in cancellables set you can remove the `_ = `. Sorry for not catching before.

Reply:
Done. Nice!